packages/@aws-cdk/aws-ec2/lib/vpc-ref.ts

```diff
@@ -62,9 +62,23 @@ export interface IVpcNetwork extends IConstruct {
 
   /**
    * Return the subnets appropriate for the placement strategy
+   *
+   * Prefer using {@link subnetIds} if you need to pass subnet IDs to a
+   * CloudFormation Resource.
    */
   subnets(placement?: VpcPlacementStrategy): IVpcSubnet[];
 
+  /**
+   * Return IDs of the subnets appropriate for the given placement strategy
+   *
+   * Requires that at least once subnet is matched, throws a descriptive
+   * error message otherwise.
+   *
+   * Prefer to use this method over {@link subnets} if you need to pass subnet
```

Comment:
This sentence strikes me as a little confusing to a user... I kinda feel like that should provide more info on why this guidance exists, and the guidance should be more clear as to how you'd be "extracting IDs" out of `subnets` otherwise.